pkg/proc/variables.go

```diff
@@ -534,8 +534,19 @@ func (g *G) System(tgt *Target) bool {
 		return false
 	}
 	switch loc.Fn.Name {
-	case "runtime.main", "runtime.handleAsyncEvent", "runtime.runfinq":
+	case "runtime.main", "runtime.handleAsyncEvent":
 		return false
+	case "runtime.runfinq":
+		s, err := goroutineEvalScope(tgt, g, 0, 0)
+		if err != nil {
+			return true
```

Comment:
The default before was to return `false`, this new change is essentially "assume it's a system goroutine unless we can prove otherwise" correct?

Reply:
Correct.

It seems to be much more rare that it is a user goroutine than a system goroutine. And the approximation in src/cmd/trace/trace.go would treat the finalizer as a system goroutine always.